bootstrap/eks/controllers/eksconfig_controller.go

```diff
@@ -251,7 +281,9 @@ func (r *EKSConfigReconciler) joinWorker(ctx context.Context, cluster *clusterv1
 		DiskSetup:                config.Spec.DiskSetup,
 		Mounts:                   config.Spec.Mounts,
 		Files:                    files,
+		ClusterCIDR:              controlPlane.Spec.NetworkSpec.VPC.CidrBlock,
```

Comment:
This should be the service CIDR of the cluster as per [documentation](https://github.com/awslabs/amazon-eks-ami/blob/v20250813/nodeadm/api/v1alpha1/nodeconfig_types.go#L52-L53):

```
// CIDR is your cluster's service CIDR block. This value is used to infer your cluster's DNS address.
CIDR string `json:"cidr,omitempty"`
```

Pushed a PR targeting your PR branch - https://github.com/AmitSahastra/cluster-api-provider-aws/pull/2.

Reply:
@AmitSahastra Could you take a look at this please?